lib/web/app/handler.go

```diff
@@ -618,10 +618,14 @@ const (
 // The URL's are formed this way to help isolate the path params reserved for the app
 // launchers route, where order and existence of previous params matter for this route.
 func makeAppRedirectURL(r *http.Request, proxyPublicAddr, hostname string, req launcherURLParams) string {
+	host := hostname
+	if req.requiresAppRedirect {
+		host = req.publicAddr
+	}
```

Comment:
```suggestion
	if req.requiresAppRedirect {
		hostname = req.publicAddr
	}
```

If you update the existing `hostname` variable instead of making a new variable with a similar name, there's less chance of us mistakenly using the wrong one later on in this function and breaking something.

Reply:
Thinking about this more I agree with how it's done now a bit more; if the variable is set to the public addr calling it `hostname` isn't very accurate anymore

Reply:
Then change the input paramater to something like `addr`.

Having two variables in scope with a similar name, where one works and the other doesn't is a recipe for failure.

Reply:
Fair enough, done

Reply:
I was hoping you would rename the function parameter, which was not done.

I don't want two variables in scope where one works and one doesn't. Let's make a single variable (which we overwrite if necessary), so there's no potential for confusion. As written, we still have `addr` and `hostname` in scope.

Reply:
Ah I see, I misunderstood. That makes sense, I renamed the `hostname` parameter to `addr`. This should make this function less confusing